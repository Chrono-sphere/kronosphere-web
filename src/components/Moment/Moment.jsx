import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Resources from './Moment.resource';
import './Moment.scss';

class Moment extends Component {
  state = {
    time: this.props.time,
    dayNumber: this.props.time.getDate(),
    monthNumber: this.props.time.getMonth() + 1,
    yearNumber: this.props.time.getFullYear(),
  }

  componentWillMount = () => setInterval(this.setCurrentTime, 60000);

  getMonth= () => {
    const months = [
      Resources.Month.january,
      Resources.Month.febuary,
      Resources.Month.march,
      Resources.Month.april,
      Resources.Month.may,
      Resources.Month.june,
      Resources.Month.july,
      Resources.Month.august,
      Resources.Month.september,
      Resources.Month.october,
      Resources.Month.november,
      Resources.Month.december,
    ];

    return months[this.state.monthNumber - 1];
  }

  getSubscript = () => {
    switch (this.state.dayNumber) {
      case 1:
      case 21:
      case 31:
        return 'st';

      case 2:
      case 22:
        return 'nd';

      case 3:
      case 23:
        return 'rd';

      default:
        return 'th';
    }
  }

  setCurrentTime = () => {
    this.setState({ time: new Date() });
  }

  render = () => (
    <div className="moment">
      <div className="moment-day-content">
        <div className="moment-day-number">
          <div className="moment-day-number-digit">
            {this.state.dayNumber}
          </div>
          <div className="moment-day-number-subscript">
            {this.getSubscript()}
          </div>
        </div>

        <div className="moment-month">
          {`${this.getMonth()} ${this.state.yearNumber}`}
        </div>
      </div>

      <div className="moment-time-content">
        {this.state.time.toLocaleTimeString([], { hour: '2-digit', minute: '2-digit', hour12: this.props.clock12 })}
      </div>
    </div>
  )
}

Moment.propTypes = {
  time: PropTypes.instanceOf(Date),
  clock12: PropTypes.bool,
};

Moment.defaultProps = {
  time: new Date(),
  clock12: true,
};

export default Moment;
