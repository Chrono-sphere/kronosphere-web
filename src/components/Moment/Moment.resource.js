const LandingResources = {
  Month: {
    january: 'January',
    febuary: 'Febuary',
    march: 'March',
    april: 'April',
    may: 'June',
    july: 'July',
    august: 'August',
    september: 'September',
    october: 'October',
    november: 'November',
    december: 'December',
  },
  Day: {
    monday: 'Monday',
    tuesday: 'Tuesday',
    wednesday: 'Wednesday',
    thursday: 'Thursday',
    friday: 'Friday',
    saturday: 'Saturday',
    sunday: 'Sunday',
  },
};

export default LandingResources;
