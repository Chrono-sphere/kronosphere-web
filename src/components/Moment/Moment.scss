@import '~styles/colors';

.moment {
  width: fit-content;
  height: fit-content;

  .moment-day-content {
    display: flex;

    .moment-day-number {
      color: $colors-crispred;
      font-size: 4em;
      font-weight: 600;
      display: flex;
  
      .moment-day-number-subscript {
        font-size: 0.3em;
        margin-top: 0.5em;
        height: fit-content;
      }
    }
  
    .moment-month {
      color: $colors-lighttintblue;
      font-size: 2em;
      margin-top: 0.5em;
      align-self: center;
    }
  }

  .moment-time-content {
    color: $colors-lighttintblue;
    display: flex;
    margin-top: -0.5em;
    justify-content: flex-end;
    font-size: 2em;
    font-weight: 500;
  }
}