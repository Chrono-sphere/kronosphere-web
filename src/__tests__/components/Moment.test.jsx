import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Moment from 'components/Moment/Moment';

describe('SimpleButton', () => {
  const date = new Date('December 9, 1992 03:24:00');

  it('should render without throwing an error', () => {
    expect(shallow(<Moment />).find('div.moment').exists()).toBe(true);
  });

  it('should set the props correctly', () => {
    const wrapper = shallow(<Moment time={date} />);
    const { className, children } = wrapper.props();
    const dayContent = children[0];
    const timeContent = children[1];
    const dayContentDay = dayContent.props.children[0];
    const dayContentMonth = dayContent.props.children[1];

    expect(className).toBe('moment');
    expect(dayContent.type).toBe('div');
    expect(timeContent.type).toBe('div');

    expect(dayContent.props.className).toBe('moment-day-content');
    expect(timeContent.props.className).toBe('moment-time-content');
    expect(timeContent.props.children).toBe('3:24 AM');

    expect(dayContentDay.props.className).toBe('moment-day-number');

    expect(dayContentMonth.props.className).toBe('moment-month');
    expect(dayContentMonth.props.children).toBe('December 1992');
  });

  it('should render deep correctly', () => {
    const tree = renderer.create(<Moment time={date} />);
    expect(tree).toMatchSnapshot();
  });
});
