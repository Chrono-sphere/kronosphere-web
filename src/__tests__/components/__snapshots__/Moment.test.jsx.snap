// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleButton should render deep correctly 1`] = `
<div
  className="moment"
>
  <div
    className="moment-day-content"
  >
    <div
      className="moment-day-number"
    >
      <div
        className="moment-day-number-digit"
      >
        9
      </div>
      <div
        className="moment-day-number-subscript"
      >
        th
      </div>
    </div>
    <div
      className="moment-month"
    >
      December 1992
    </div>
  </div>
  <div
    className="moment-time-content"
  >
    3:24 AM
  </div>
</div>
`;
